homeassistant/components/frontend/storage.py

```diff
@@ -83,6 +89,62 @@ def __init__(self, hass: HomeAssistant, user_id: str) -> None:
         )
 
 
+@singleton.singleton(DATA_SYSTEM_STORAGE, async_=True)
+async def async_system_store(hass: HomeAssistant) -> SystemStore:
+    """Access the system store."""
+    store = SystemStore(hass)
+    await store.async_load()
+    return store
+
+
+class SystemStore:
+    """System store for frontend data."""
+
+    def __init__(self, hass: HomeAssistant) -> None:
+        """Initialize the system store."""
+        self._store = _SystemStore(hass)
+        self.data: dict[str, Any] = {}
+        self.subscriptions: dict[str | None, list[Callable[[], None]]] = {}
+
+    async def async_load(self) -> None:
+        """Load the data from the store."""
+        self.data = await self._store.async_load() or {}
+
+    async def async_set_item(self, key: str, value: Any) -> None:
+        """Set an item and save the store."""
+        self.data[key] = value
+        await self._store.async_save(self.data)
```

Comment:
Is this important enough that we need to do immediate saving? with stores we often call delay_save, so the whole function becomes a callback and if a user changes something within 60 seconds again, it won't trigger a second write.

Reply:
(delayed writes are always done when HA shuts down)